lib/components.go

```diff
@@ -0,0 +1,76 @@
+package lib
+
+import (
+	"sync"
+
+	"github.com/opencontrol/compliance-masonry/lib/components"
+	"github.com/opencontrol/compliance-masonry/lib/common"
+	"fmt"
+)
+
+// componentsMap struct is a thread-safe structure mapping for components
+type componentsMap struct {
+	mapping map[string]common.Component
+	sync.RWMutex
+}
+
+// newComponents creates an instance of Components struct
+func newComponents() *componentsMap {
+	return &componentsMap{mapping: make(map[string]common.Component)}
+}
+
+// add adds a new component to the component map
+func (components *componentsMap) add(component common.Component) {
+	components.Lock()
+	components.mapping[component.GetKey()] = component
+	components.Unlock()
+}
+
+// Get retrieves a new component from the component map
+func (components *componentsMap) Get(key string) common.Component {
+	components.RLock()
+	defer components.RUnlock()
+	return components.mapping[key]
+}
+
+// CompareAndAdd compares to see if the component exists in the map. If not, it adds the component.
+// This function is thread-safe.
+func (components *componentsMap) CompareAndAdd(component common.Component) bool {
+	components.Lock()
+	defer components.Unlock()
+	added := false
+	if _, exists := components.mapping[component.GetKey()]; !exists {
```

Comment:
While I know this isn't unusual for Go, I'm personally not a fan of cramming assignment plus a check into a conditional. I favor one thing per line.